File: app/src/main/java/com/example/david/lists/util/UtilWidgetKeys.java
package com.example.david.lists.util;

import android.content.Context;

import com.example.david.lists.R;

public class UtilWidgetKeys {
    private UtilWidgetKeys() {
    }

    public static String getSharedPrefName(Context context) {
        return context.getString(R.string.widget_shared_prefs_name);
    }

    public static String getSharedPrefKeyId(Context context, int appWidgetId) {
        return context.getString(R.string.widget_key_shared_pref_user_list_id, appWidgetId);
    }

    public static String getSharedPrefKeyTitle(Context context, int appWidgetId) {
        return context.getString(R.string.widget_key_shared_pref_user_list_title, appWidgetId);
    }
}

File: app/src/main/java/com/example/david/lists/widget/common/UtilWidgetKeys.kt
package com.example.david.lists.widget.common

import com.example.david.lists.R

class UtilWidgetKeys(private val getStringRes: (Int) -> String,
                     private val getStringResArg: (Int, Int) -> String) {

    fun getSharedPrefName() =
            getStringRes(R.string.widget_shared_prefs_name)

    fun getSharedPrefKeyId(appWidgetId: Int) =
            getStringResArg(R.string.widget_key_shared_pref_user_list_id, appWidgetId)

    fun getSharedPrefKeyTitle(appWidgetId: Int) =
            getStringResArg(R.string.widget_key_shared_pref_user_list_title, appWidgetId)
}
